test/test_prototype_transforms_consistency.py

```diff
@@ -642,6 +649,74 @@ def test_call_consistency(config, args_kwargs):
     )
 
 
+@consistency_configs
+def test_jit_consistency(config, args_kwargs):
+    args, kwargs = args_kwargs
+
+    prototype_transform_eager = config.prototype_cls(*args, **kwargs)
+    legacy_transform_eager = config.legacy_cls(*args, **kwargs)
+
+    try:
+        legacy_transform_scripted = torch.jit.script(legacy_transform_eager)
+    except Exception as exc:
+        msg = str(exc)
+
+        # Some of the transform parameter variations cannot be used while scripting. For example, `Resize(size=1)`
+        # is not scriptable. One has to use `Resize(size=[1])` instead. To avoid creating a second set of
+        # parameters, we just abort the JIT call check in such a case.
+        if (
+            re.search(r"Expected a value of type 'List\[int\]' for argument '\w+' but instead found type 'int'", msg)
+            is not None
+        ):
+            return
+        if "Expected a value of type 'Union[float, int]' for argument 'fill'" in msg:
+            return
+        # In some cases trying to script the same transformation multiple times with parameters that are not supported,
+        # `torch.jit.script` will emit a cryptic error message rather than one like the ones above.
+        if "Can't redefine method: forward on class:" in msg:
+            return
+
+        # This error happens when `torch.jit.script` hits an unguarded `_log_api_usage_once`. Since that means that
+        # the transform doesn't support scripting in general, we abort the JIT call check.
+        if "'Any' object has no attribute or method '__module__'" in msg:
+            return
+
+        # If we hit some other error, we fail loudly to avoid a false sense of security
+        raise pytest.UsageError("The legacy transform cannot be scripted!") from exc
+
+    try:
+        prototype_transform_scripted = torch.jit.script(prototype_transform_eager)
+    except Exception as exc:
+        raise AssertionError("The prototype transform cannot be scripted!") from exc
+
+    for image in make_images(**config.make_images_kwargs):
+        image = image.as_subclass(torch.Tensor)
+        image_repr = f"[{tuple(image.shape)}, {str(image.dtype).rsplit('.')[-1]}]"
+
+        try:
+            torch.manual_seed(0)
+            output_legacy_scripted = legacy_transform_scripted(image)
+        except Exception as exc:
+            raise AssertionError(
+                f"Calling the scripted legacy transform with {image_repr} raised the error above!"
+            ) from exc
```

Comment:
Do we really need these try/except? I understand the intention is a good one, and definitely worth it for user-facing utilities. But this is developer-facing code; personally I feel like the try/except + custom error message adds no value and mostly just obfuscates the traceback.

Reply:
Another thing that makes me sceptical of their value: these custom error messages are not always accurate and can be misleading. While working on  https://github.com/pytorch/vision/pull/7122 I had completely messed up the transforms code (parameter typo, stuff like that). Yet the failure I was seeing was

```
                f"Transforming a image datapoint with shape {image_repr} failed in the prototype transform with "
                f"the error above. This means there is a consistency bug either in `_get_params` or in the "
                f"`datapoints.Image` path in `_transform`."
```

But it was not really a consistency bug, it was just me writing garbage code. I'd rather get a clear `SyntaxError` straight at the bottom of the traceback rather than the message above which was misleading.

Again I agree the intention is right - but I think this is mostly valuable for user-facing APIs.

Reply:
Do you want to rip them out for everything not related to this PR or shall I?

Reply:
We can probably leave the pre-existing ones for now, and address them when we go over the tests later